### ISSUE TITLE
How to use it on VM ParrotOS

### ISSUE THREAD

**Issue by adcoding** [JustAnotherArchivist/snscrape#1043]
Hello!
I am new to this tool.
I've been trying to use this library but there no way I can make it work.

I set up a virtual machine with ParrotOS.
But all I can get is: Illegal instruction.

![image](https://github.com/JustAnotherArchivist/snscrape/assets/50012804/b0d117ea-17a7-4627-aea2-1295563b2a2f)

![image](https://github.com/JustAnotherArchivist/snscrape/assets/50012804/91162fdf-b579-4b60-815f-c617dd5cc800)

Thank you for your help.

**Comment by JustAnotherArchivist** [comment]
This is the first time I hear of Parrot OS, so I'm afraid I have no idea what might be wrong there. snscrape itself is pure Python code though, and 'Illegal instruction' is something that would be caused by machine code. So I think the issue lies elsewhere. Given that you're able to run `pip`, Python seems to likely be fine. So I would guess that one of the other packages wasn't compiled for the correct CPU architecture or similar. As a starting point, perhaps try to import the dependencies of snscrape individually to see which one breaks.

**Comment by adcoding** [comment]
Yeah I have tried to install the dependencies individually but I still get the same error.
I also tried on Ubuntu and it is the same so most likely is something related to the way I set up my virtual machine.
No clue tho.

**Comment by TheTechRobo** [comment]
Does it work outside the VMs?

Also try *importing* each dependency individually.

**Comment by adcoding** [comment]
now I'm trying on windows 11. (python 3.12.2)
and I have installed the dependencies individually.

![image](https://github.com/JustAnotherArchivist/snscrape/assets/50012804/1ec9aab4-87b1-4398-bd23-08724b11aafe)

I also get this error:
![image](https://github.com/JustAnotherArchivist/snscrape/assets/50012804/f258e506-289d-4089-903e-e2ac82f2b903)

![image](https://github.com/JustAnotherArchivist/snscrape/assets/50012804/c9473be0-8e77-48fa-b7a8-10431fb7398f)

when I tried to run `snscrape --help` , I got this error:
![Screenshot 2024-04-04 105040](https://github.com/JustAnotherArchivist/snscrape/assets/50012804/65d01da4-6752-400b-b231-4b0529a482be)
